impl/client/src/main/javascript/web/pentaho/data/Attribute.js

```diff
@@ -116,27 +116,50 @@ define([
        */
       this.type = type;
 
-      var isDiscrete = type !== "number" && type !== "date";
-      if(!isDiscrete && spec.isDiscrete != null)
-        isDiscrete = !!spec.isDiscrete;
+      var isContinuous;
+      if(type === "string" || type === "boolean") {
```

Comment:
This default takes precedence even if `spec.isContinuous` is defined. Is that intended? (I understand it makes sense to force as categorical something continuous but not the other way around... my question is if we should "auto-correct" the spec choice at this level)

Reply:
String is inherently categorical. The only way it could be seen as continuous is if some conversion could be performed on it. Because the data table offers no means for such conversion, then it is necessarily categorical.
With, boolean, we could also see it in a continuous numeric scale, if we would map `false` to `0` and `true` to `1`, for example. Still, a conversion is being assumed.
I don't see this as an automatic correction, but more like the `spec.isContinuous` option only applies (ignored otherwise) to types which _can be_ continuous, and for some types, we know they cannot be.